plugin/src/adapter34/kotlin/org/jlleitschuh/gradle/ktlint/reporter/SerializableReportersProviderLoader.kt

```diff
@@ -0,0 +1,77 @@
+package org.jlleitschuh.gradle.ktlint.reporter
+
+import com.pinterest.ktlint.core.Reporter
+import com.pinterest.ktlint.core.ReporterProvider
+import java.io.File
+import java.io.ObjectInputStream
+import java.util.ServiceLoader
+
+class SerializableReportersProviderLoader :
+    ReportersLoaderAdapter<Reporter, SerializableReporterProvider, Ktlint34Reporter, Ktlint34ReporterProvider> {
+    override fun loadAllReporterProviders(): List<ReporterProviderWrapper<SerializableReporterProvider>> = ServiceLoader
+        .load(ReporterProvider::class.java)
+        .toList().map {
+            ReporterProviderWrapper(it.id, SerializableReporterProvider(it))
+        }
```

Comment:
Good use of service loaders IMHO!

Reply:
I can't take credit, this code was already here :)